File: dayTwo/python/book_suggestion_system_main.py
from book_suggestion_system import BookSuggestionSystem

book_suggestion_system = BookSuggestionSystem()

while True:

    print("""Welcome to the Book Suggestion System!
1. Get Suggestions
2. Add Book
3. Remove Book
4. Update Book
5. Show Books""")

    operation = int(input("Enter operation: "))

    print()

    match operation:

        case 1:
            option = ""

            try:
                while True:

                    book_recommendation = book_suggestion_system.suggest_book()
                    print(book_recommendation)

                    option = input("Would you like to get another suggestion? (yes/no): ")

                    if option.strip().lower() != "yes":
                        break

                    print()

            except Exception as e:
                print(e)

        case 2:
            try:
                book_title = input("Enter the book title: ")

                book_suggestion_system.add_book(book_title)

                print("Book added successfully!")

            except Exception as e:
                print(e)

        case 3:
            try:
                book_title = input("Enter the book title: ")

                book_suggestion_system.remove_book(book_title)

                print("Book removed successfully!")

            except Exception as e:
                print(e)

        case 4:
            try:
                old_book_title = input("Enter old book title: ")
                new_book_title = input("Enter new book title: ")

                book_suggestion_system.update_book(
                    old_book_title,
                    new_book_title
                )
                print("Book updated successfully!")

            except Exception as e:
                print(e)

        case 5:
            book_list = book_suggestion_system.get_book_list()

            print("All Books")

            if len(book_list) == 0:
                print("Nothing to see here yet!")
            else:
                for index, book in enumerate(book_list):
                    print(f"{index + 1}. {book}")

        case _:
            print("Invalid operation!!!")

    print()

File: dayTwo/javaScript/bookSuggestionSystemMain.js
const BookSuggestionSystem = require('./bookSuggestionSystem');

const prompt = require('prompt-sync')();

const bookSuggestionSystem = new BookSuggestionSystem();

let keepSuggesting = true;

while (keepSuggesting) {
    console.log(`Welcome to the Book Suggestion System!
1. Get Suggestions
2. Add Book
3. Remove Book
4. Update book
5. Show books`)
    let operation = Number(prompt("Enter operation: "))

    console.log()
    switch (operation) {
        case 1:
            let option = "";

            try {
                do {
                    let bookRecommendation = bookSuggestionSystem.suggestBook();
                    console.log(bookRecommendation);

                    option = prompt("Would you like to get another suggestion? (yes/no): ");
                    if (option.trim().toLowerCase() === "yes")
                        console.log("");

                } while(option.trim().toLowerCase() === "yes");
            } catch(error) {
                console.log(error.message);
            }
            break;
        case 2:
            try {
                let bookTitle = prompt("Enter the book title: ");
                bookSuggestionSystem.addBook(bookTitle);
                console.log("Book added successfully!");
            } catch(error) {
                console.log(error.message);
            }
            break;
        case 3:
            try {
                let bookTitle = prompt("Enter the book title: ");
                bookSuggestionSystem.removeBook(bookTitle);
                console.log("Book removed successfully!");
            } catch(error) {
                console.log(error.message);
            }
            break;
        case 4:
            try {
                let oldBookTitle = prompt("Enter old book title: ");
                let newBookTitle = prompt("Enter new book title: ");

                bookSuggestionSystem.updateBook(oldBookTitle, newBookTitle);
                console.log("Book updated successfully!");
            } catch(error) {
                console.log(error.message);
            }
            break;
        case 5:
            let bookList = bookSuggestionSystem.getBookList();
            console.log("All Books");

            if (bookList.length == 0) {
                console.log("Nothing to see here yet!");
                break;
            }

            let index = 1;
            for (let book of bookList) {
                console.log(`${index}. ${book}`);
                index++;
            }

            break;
        default:
            console.log("Invalid operation!!!");
    }

    console.log()
}